torch/csrc/jit/script/parser.cpp

```diff
@@ -649,19 +649,21 @@ struct ParserImpl {
   TreeRef parseClassLike() {
     L.expect(TK_CLASS_DEF);
     const auto name = parseIdent();
+    Maybe<Ident> superclass = Maybe<Ident>::create(name.range());
     if (L.nextIf('(')) {
       // Only support inheriting from NamedTuple right now.
-      if (L.cur().kind == TK_IDENT && L.cur().text() == "NamedTuple") {
-        L.next();
+      auto id = L.expect(TK_IDENT);
+      if (id.text() == "NamedTuple") {
         return parseNamedTuple(name);
       } else {
-        L.reportError("Inheritance is not supported for TorchScript classes");
+        superclass = Maybe<Ident>::create(id.range, Ident::create(id.range, id.text()));
```

Comment:
We should probably serialize with fully qualified names for modules, meaning this will be torch.nn.Module not `Module` (and hence not an identifier). Why? What happens when a user manages to call their own module Module. Since we do not allow inheritance now, it can't happen, but I can imagine it happening in the future.